mcs/class/corlib/ReferenceSources/EncodingTable.cs

```diff
@@ -747,14 +745,18 @@ internal static partial class EncodingTable
         [SecurityCritical]
         unsafe internal static InternalCodePageDataItem* codePageDataPtr = GetCodePageData();
 */
+
+        // Protects the two maps below
+        static object mutex = new Object ();
+
         //
         // This caches the mapping of an encoding name to a code page.
         //
-        private static Hashtable hashByName = Hashtable.Synchronized(new Hashtable(StringComparer.OrdinalIgnoreCase));
+        private static Dictionary<string, int> hashByName = new Dictionary<string, int> (StringComparer.OrdinalIgnoreCase);
```

Comment:
The two dictionaries were independent before the refactoring. Better to lock on `hashByName.SyncRoot`?

Reply:
Fixed.